packages/lending/src/protocols/morphoblue/lending-protocol.ts

```diff
@@ -0,0 +1,229 @@
+import { BigNumber } from 'ethers';
+import BigNumberJS from 'bignumber.js';
+import { BorrowObject, Market, RepayParams, SupplyObject } from 'src/protocol.type';
+import { DISPLAY_NAME, ID, configMap, getContractAddress, getMarket, marketMap, supportedChainIds } from './configs';
+import { IrmInterface, MarketParamsStruct, MarketStruct } from './contracts/Irm';
+import { Irm__factory, Morpho, Morpho__factory, Oracle__factory, PriceFeed__factory } from './contracts';
+import { MorphoInterface } from './contracts/Morpho';
+import { OracleInterface } from './contracts/Oracle';
+import { Portfolio } from 'src/protocol.portfolio';
+import { PriceFeedInterface } from './contracts/PriceFeed';
+import { Protocol } from 'src/protocol';
+import { SECONDS_PER_YEAR } from '@aave/math-utils';
+import * as apisdk from '@protocolink/api';
+import * as common from '@protocolink/common';
+
+export class LendingProtocol extends Protocol {
+  readonly id = ID;
+
+  static readonly markets = supportedChainIds.reduce((accumulator, chainId) => {
+    for (const marketId of Object.keys(marketMap[chainId])) {
+      accumulator.push({ id: marketId, chainId });
+    }
+    return accumulator;
+  }, [] as Market[]);
+
+  private _morpho?: Morpho;
+
+  get morpho() {
+    if (!this._morpho) {
+      this._morpho = Morpho__factory.connect(getContractAddress(this.chainId, 'Morpho'), this.provider);
+    }
+    return this._morpho;
+  }
+
+  private _morphoIface?: MorphoInterface;
+
+  get morphoIface() {
+    if (!this._morphoIface) {
+      this._morphoIface = Morpho__factory.createInterface();
+    }
+    return this._morphoIface;
+  }
+
+  private _priceFeedIface?: PriceFeedInterface;
+
+  get priceFeedIface() {
+    if (!this._priceFeedIface) {
+      this._priceFeedIface = PriceFeed__factory.createInterface();
+    }
+    return this._priceFeedIface;
+  }
+
+  private _oracleIface?: OracleInterface;
+
+  get oracleIface() {
+    if (!this._oracleIface) {
+      this._oracleIface = Oracle__factory.createInterface();
+    }
+    return this._oracleIface;
+  }
+
+  private _irmIface?: IrmInterface;
+
+  get irmIface() {
+    if (!this._irmIface) {
+      this._irmIface = Irm__factory.createInterface();
+    }
+    return this._irmIface;
+  }
+
+  getMarketName(id: string) {
+    const { loanToken, collateralToken } = getMarket(this.chainId, id);
+    return `${DISPLAY_NAME} ${loanToken.symbol}(${collateralToken.symbol} collateral)`;
```

Comment:
When would we use this function? For me, the market info is still confusing if I want to know more than the tokens.

Reply:
We use this function `getMarketName` on the frontend to display the title of the market, such as `Aave V3` `Compound V3 USDC.e` `Morpho Blue USDC(ETH collateral)`

Reply:
In that case, it might be necessary to show some info about the marketId, or it is acceptable to have the same name?